### PR TITLE
Geoip and cloudflare updates

### PR DESCRIPTION
**Pull request by jencub**
Updates to GeoIP due to changes in download format
Add cron job for GeoIP

Ref:  https://jira.greenpeace.org/browse/PLANET-4643

Add cron job for Cloudflare

Ref:  https://jira.greenpeace.org/browse/PLANET-4606

### CODE REVIEW

File: src/planet-4-151612/openresty/bin/nginx_configure_cloudflare_cron.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+set -e
+
+# Setup daily cron job to get CloudFlare updated ips
+# by running the nginx_configure_cloudflare.sh script with mods
+
+CLOUDFLARE_DAILY_CRON_FILE_PATH="/etc/cron.daily/nginx_update_cloudflare_ips"
```
Comment (27Bslash6) [inline src/planet-4-151612/openresty/bin/nginx_configure_cloudflare_cron.sh]: Perhaps test against a global variable, and enable disable the cron job according to `CLOUDFLARE_ENABLED` instead? this way the jobs only run when we need them.

```bash
if [ "${CLOUDFLARE_ENABLED,,}" != "true" ]
then
  chmod 644 "$CLOUDFLARE_DAILY_CRON_FILE"
else 
  chmod 755 "$CLOUDFLARE_DAILY_CRON_FILE"
fi
```

Another question, have you confirmed the scripts don't run when not executable?

Unit tests alongside new functionality would be preferable.

File: src/planet-4-151612/openresty/etc/GeoIP.conf [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+# https://dev.maxmind.com/geoip/geoipupdate/.
+# Account is currently registered with jhasenau@greenpeace.org
 
-# Include one or more of the following edition IDs:
-# * GeoLite2-City - GeoLite 2 City
-# * GeoLite2-Country - GeoLite2 Country
-# For geoipupdate versions earlier than 2.5.0, use ProductIds here instead of EditionIDs.
-ProductIds GeoLite2-City GeoLite2-Country
+# `AccountID` is from your MaxMind account.
+AccountID 146520
```
Comment (27Bslash6) [inline src/planet-4-151612/openresty/etc/GeoIP.conf]: Is hard-coding these secrets in publically accessible VCS a good idea?

These configuration files should likely be generated with dockerize, and the secrets injected by env vars on startup.

Will require modifying Helm charts (planet4 and static), CI environment, etc.

File: src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+echo "# CloudFlare IP Ranges" > $CLOUDFLARE_IP_RANGES_FILE_PATH_TODAY
+echo "# Generated at $(date) by $0" >> $CLOUDFLARE_IP_RANGES_FILE_PATH_TODAY
+echo "" >> $CLOUDFLARE_IP_RANGES_FILE_PATH_TODAY
+awk '{ print "set_real_ip_from " $0 ";" }' $CLOUDFLARE_IPSV4_LOCAL_FILE >> $CLOUDFLARE_IP_RANGES_FILE_PATH_TODAY
+awk '{ print "set_real_ip_from " $0 ";" }' $CLOUDFLARE_IPSV6_LOCAL_FILE >> $CLOUDFLARE_IP_RANGES_FILE_PATH_TODAY
+# This real_ip_header value is already declared by nginx_configure_real_ip.sh
+#echo "real_ip_header CF-Connecting-IP;" >> $CLOUDFLARE_IP_RANGES_FILE_PATH_TODAY
+echo "" >> $CLOUDFLARE_IP_RANGES_FILE_PATH_TODAY
+
+chown $WWW_USER:$WWW_GROUP $CLOUDFLARE_IP_RANGES_FILE_PATH_TODAY
```
Comment (27Bslash6) [inline src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips]: Replace WWW_USER:WWW_GROUP with APP_USER:APP_GROUP, remove assignment above.

File: src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#  THE SOFTWARE.
+
```
Comment (27Bslash6) [inline src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips]: Add `set -euo pipefail`

File: src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+CLOUDFLARE_IP_RANGES_FILE_PATH="/etc/nginx/conf.d/cloudflare-ips.conf"
+WWW_GROUP=${APP_GROUP}
+WWW_USER=${APP_USER}
+
+CLOUDFLARE_IPSV4_REMOTE_FILE="https://www.cloudflare.com/ips-v4/"
+CLOUDFLARE_IPSV6_REMOTE_FILE="https://www.cloudflare.com/ips-v6/"
+CLOUDFLARE_IPSV4_LOCAL_FILE="/tmp/cloudflare-ips-v4"
+CLOUDFLARE_IPSV6_LOCAL_FILE="/tmp/cloudflare-ips-v6"
+
+if [ -f /usr/bin/fetch ];
```
Comment (27Bslash6) [inline src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips]: We don't have fetch in our environment

**Comment by jencub** [comment]
- cron jobs cant have extensions in the cron.xxx folders
- the fetch was already there .. will remove it
- the secrets dont avail anything but the free download per the new geoip
requirements ... but if the env config needs doin should probs get them
requested again under a generic email  account
- yes we should be using the enabled tag ..  but wasnt sure if this could
be fixed post deployment to get these fixes in ...

**Comment by 27Bslash6** [comment]
Okay, if cron.x folders don't support file extensions, then I'd recommend the script create or delete a symlink to a .sh depending on value.

While we could fix the logic in post, I'd prefer to see newly committed code honouring ENV configuration at the outset.  Tests can, but really should not, wait if it's critical to get this into production asap.

File: src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips (hunk 4 above)
Comment (jencub) [reply src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips]: Added

File: src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips (hunk 3 above)
Comment (jencub) [reply src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips]: done

File: src/planet-4-151612/openresty/bin/nginx_configure_cloudflare_cron.sh (hunk 1 above)
Comment (jencub) [reply src/planet-4-151612/openresty/bin/nginx_configure_cloudflare_cron.sh]: - added global variable to real ip and cloudflare config file
- added unit tests for cloudflare and cron jobs

File: src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips (hunk 5 above)
Comment (jencub) [reply src/planet-4-151612/openresty/etc/cron.daily/nginx_update_cloudflare_ips]: fetch removed

**Pull request merged**